### PR TITLE
Add JSON Structured Logging to data plane components

### PR DESCRIPTION
**Pull request by pierDipi**
## Proposed Changes

- Add JSON Structured Logging to data plane components

### CODE REVIEW

File: data-plane/config/100-config-logging.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kafka-broker-config-logging
+  namespace: knative-eventing
+data:
+  config.xml: |
```
Comment (slinkydeveloper) [inline data-plane/config/100-config-logging.yaml:7]: an xml inside a yaml FTW

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java [hunk 2]
```diff
@@ -52,6 +55,13 @@
    */
   public static void main(final String[] args) {
 
+    // HACK HACK HACK
+    // maven-shade-plugin doesn't include the LogstashEncoder class, neither by specifying the
+    // dependency with scope `provided` nor `runtime`, and adding include rules to
+    // maven-shade-plugin.
+    // Instantiating an Encoder here we force it to include the class.
```
Comment (slinkydeveloper) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java:62]: What about using include directives? http://maven.apache.org/plugins/maven-shade-plugin/examples/includes-excludes.html
So you force to include that class

Comment (pierDipi) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java:62]: I tried never got it working :cry:

Comment (slinkydeveloper) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java:62]: Ok let's keep it this way and we'll figure out later

**Pull request merged**